### DISCUSSION TITLE
Allow swappable translation engines

### DISCUSSION THREAD

**Discussion by sr229** [Earu/EasyChat#116] [Ideas]
I could suggest replacing Yandex with DeepL but considering the situation I think it's better if you can choose between different translations.

**Comment by Cynosphere** [comment]
[SimplyTranslate](https://simple-web.org/projects/simplytranslate.html) is probably the best idea here. All that would be needed is a config option for setting instance URL.

**Comment by Earu** [comment]
So this is not relevant anymore everything has been migrated to use ollama and gemma3

**Reply by Cynosphere** [reply]
This is kind of an accessibility concern because not everyone wants to (or can't, GMod still runs on ancient hardware) run a LLM locally just to use translation. Maybe make it work serverside as well so it can be optionally setup by the server owners as well?

**Reply by Earu** [reply]
No that wont work, most servers dont have GPUs, it would delay messages by a considerable amount of time while also putting stress on the whole machine the server is running on

**Reply by Earu** [reply]
I do understand the hardware argument but I do believe this is the best compromise there is for translation, I don't really want to pay for translation services. I'm not rich enough not to care about potentially hundreds of users using this every month for free.

**Reply by Earu** [reply]
What I can do though, is add some setting to set where the ollama server is hosted (and which port), so if you want to run a server that runs it you can do it but be warned, performance will be abysmal.
